src/mlpack/tests/load_save_test.cpp

```diff
@@ -2527,3 +2527,33 @@ TEST_CASE("LoadCSVHeaderTest", "[LoadSaveTest]")
   REQUIRE(headers.at(2) == "c");
   REQUIRE(headers.at(3) == "d");
 }
+
+TEST_CASE("LoadVectorCSVFiles", "[LoadSaveTest]")
+{
+  std::vector<std::string> files = {"f0.csv", "f1.csv", "f2.csv", "f3.csv",
+      "f3.csv", "f4.csv", "f5.csv", "f6.csv", "f7.csv", "f8.csv", "f9.csv"};
+
+  arma::mat dataset;
+  data::TextOptions opts;
+  REQUIRE(data::Load(files, dataset, opts)
+      == true);
+
+  REQUIRE(dataset.n_rows == 5);
+  REQUIRE(dataset.n_cols == 110);
+}
+
+TEST_CASE("LoadVectorCSVFilesNoTranspose", "[LoadSaveTest]")
+{
+  std::vector<std::string> files = {"f0.csv", "f1.csv", "f2.csv", "f3.csv",
+      "f3.csv", "f4.csv", "f5.csv", "f6.csv", "f7.csv", "f8.csv", "f9.csv"};
+
+  arma::mat dataset;
+  data::TextOptions opts;
+  opts.NoTranspose() = true;
+  opts.Fatal() = true;
+  REQUIRE(data::Load(files, dataset, opts)
+      == true);
+
+  REQUIRE(dataset.n_rows == 110);
+  REQUIRE(dataset.n_cols == 5);
+}
```

Comment:
Do you think we could also add tests for these situations?

 * `files` is empty
 * `files` has length 1 (e.g. only loading one file)
 * Two files have different numbers of columns
 * CSVs have headers, but they don't match
 * CSVs have headers that don't match... but we load with `MissingToNan()` and `HasHeaders() = false`... so it should still load just fine

Reply:
I will test the last one, but I think it is not possible since we do not support loading a file with a header and ignoring it. As we recently removed the support for loading them as 0

Reply:
All these tests look awesome.  Thanks for adding them.